Fix api issue with new configkey validation

lucs7 opened the pull request:
This pull request refactors how API group configuration is handled and improves error handling and logging consistency throughout the codebase. The main changes include centralizing configuration key usage, improving error messages, and standardizing log level values.

**Configuration and Error Handling Improvements:**

* Refactored all API group configuration lookups in `Web/Services/index.php` to use constants from `ConfigKeys` instead of string literals, ensuring consistency and reducing the risk of typos. [[1]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L101-R102) [[2]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L119-R121) [[3]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L142-R143) [[4]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L163-R164) [[5]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L180-R181) [[6]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L196-R197) [[7]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L210-R219) [[8]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L227-R228) [[9]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L252-R254)
* Updated the `GetConfigGroup` function to accept either a string or array config key, improved its documentation, and changed error handling to throw a `RuntimeException` with a clear message if the group cannot be found, instead of calling `die()`.
* Moved the API enabled check from the global scope to the Slim `slim.before.dispatch` hook, using `$app->halt()` with a `503 Service Unavailable` status code and a clear message when the API is disabled. [[1]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81L25-L27) [[2]](diffhunk://#diff-c966d0211b9c12b49dd812601ba838967e63b5d5019eb481d3ea331d2b542c81R46-R49)
* Updated the call to `AddCategory` in `RegisterAttributes` to use a named parameter for clarity.

**Logging Consistency:**

* Standardized logging level values in `ConfigKeys` to use lowercase (`debug`, `info`, `warning`, `error`) instead of uppercase, and updated descriptions accordingly.
* Ensured the log folder path in `Log.php` is always normalized by removing any trailing slash.

**HTTP Status Codes:**

* Added a `SERVICE_UNAVAILABLE` (503) constant to the `RestResponse` class for clearer and more maintainable HTTP responses.

Fixes #805

lucs7 committed:
fix: GetConfigGroup and ConfiKeys in API

lucs7 committed:
refactor: replaced die for proper server response

lucs7 committed:
fix: changed wrong auth details response code to 401

lucs7 committed:
test: update AuthenticationWebServiceTest to expect response codes

JohnVillalovos commented on lib/Config/ConfigKeys.php:
```diff
@@ -414,16 +414,16 @@ class ConfigKeys
+            'warning' => 'warning',
+            'error' => 'error',
+            'info' => 'info',
+            'debug' => 'debug',
```
This seems like a breaking change?

What happens for users who are currently using the uppercase versions?

lucs7 replied:
Lets say its a bug fix as we always had it lowercase and i just mistyped it
The logic is now always converting to lowercase which is used in Log.php for the case check

lucs7 committed:
feat: improved error handling for missing configkey and api
failure of api before slim was initiated resulted in 200 response which is now mitigated by exception_handler

JohnVillalovos reviewed (approved):
Thanks @lucs7
